apis/v1beta1/providerconfig_types.go

```diff
@@ -39,6 +39,11 @@ type ProviderCredentials struct {
 	// +kubebuilder:validation:Enum=None;Secret;InjectedIdentity;Environment;Filesystem
 	Source xpv1.CredentialsSource `json:"source"`
 
+	// AssumeRoleARN to assume with provider credentials
+	// +immutable
+	// +optional
+	AssumeRoleARN string `json:"assumeRoleARN,omitempty"`
```

Comment:
It seems like this option is a sub-configuration of `InjectedIdentity` option. Would it make sense to have a `InjectedIdentity *InjectedIdentity` field, similar to `SecretRef *SecretKeySelector` and have this option under that struct?

Reply:
currently yes - but in future we can also use AssumeRoleARN for more - what do you think ?

Reply:
Do you mean another top level option? Right now, it is programatically enabled only if you use `InjectedIdentity`, so I'd say that it could be a different field in under those other top level options if they decide to need it.

Reply:
@haarchri when you say it can be used for more, I assume you are referring to the fact that you can use the other credential sources to also assume a role -- is that correct?

Reply:
yes

Reply:
@haarchri @muvaf with that being the case, I almost wonder if it should live outside of the `credentials` struct, similar to how `provider-gcp` has `projectID` (ref https://doc.crds.dev/github.com/crossplane/provider-gcp/gcp.crossplane.io/ProviderConfig/v1beta1@v0.19.0#spec-projectID). A helpful categorization might be:
- `credentials` is how you actually **acquire** the base credentials
- anything outside of that in `spec` is how you **use** the credentials you acquired

What do y'all think about that?

Reply:
for me fine - i can change the pr for it

Reply:
@hasheddan checked & changed all suggestions ;)  - now i will start testing a few cases after chage

Reply:
I submitted my comment and now I can 😒 
> For some reason I can't comment on the conversation that @hasheddan added to. I think it's a great idea. @haarchri is it possible today make the code actually use that AssumeRoleARN field no matter what credential type you use if it's given?